### PR TITLE
adding confidence and status checks for pyth

### PR DESCRIPTION
**Pull request by nope-finance**
ran tests for token-lending locally (i know the git hooks dont work not but are resolved in other branch)

### CODE REVIEW

**Comment by DaSichuan** [comment]
The title says that we are doing status checks as well?
I don't see any status checks in the code other than the error messages

File: token-lending/program/src/processor.rs [hunk 1]
```diff
@@ -2126,6 +2134,20 @@ fn get_pyth_price(pyth_price_info: &AccountInfo, clock: &Clock) -> Result<Decima
...
+    let conf = pyth_price.agg.conf;
+
+    let confidence_ratio: u64 = 10;
+    // Perhaps confidence_ratio should exist as a per reserve config
+    // 100/confidence_ratio = maximum size of confidence range as a percent of price
+    // confidence_ratio of 10 filters out pyth prices with conf > 10% of price
+    if conf.checked_mul(confidence_ratio).unwrap() > price {
+        msg!(
+            "Oracle price confidence is too wide: {}",
+            pyth_price.agg.status as u8
```
Comment (DaSichuan) [inline token-lending/program/src/processor.rs]: Isn't this always just going to be `pyth::PriceStatus::Trading`?

Seems better to log conf and price

**Review by DaSichuan — CHANGES_REQUESTED** [review]

File: token-lending/program/src/processor.rs (hunk 1 above)
Comment (nope-finance) [reply token-lending/program/src/processor.rs]: ah yeah meant to do that

**Comment by oxrooter** [comment]
lgtm

**Pull request merged**